### PR TITLE
Generate doxygen documentation [9378]

### PR DESCRIPTION
**Pull request by JLBuenoLopez-eProsima**
Modify CMake to use BUILD_DOCUMENTATION option.
Correct several bugs in doxygen parameters
Modify README.html.in and doxyfile.in to show specific information about Fast CDR
Include several utils needed by doxigen to produce the documentation

Fast-CDR-v1_0_15.pdf: eProsima/all-docs#17

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/33/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/34/)

File: CMakeLists.txt [hunk 1]
```diff
@@ -178,6 +184,110 @@ if(EPROSIMA_BUILD_TESTS AND IS_TOP_LEVEL AND NOT EPROSIMA_INSTALLER)
...
+    configure_file(doc/README.html.in ${README_LOCATION}/README.html @ONLY IMMEDIATE)
+
+    ### ReadTheDocs ########################
+#    if(NOT CHECK_DOCUMENTATION)
+#        add_custom_target(readthedocs
+#            COMMAND "${WGET_EXE}" "https://media.readthedocs.org/htmlzip/eprosima-fast-rtps/v${PROJECT_VERSION}/eprosima-fast-rtps.zip"
+#            COMMAND "${UNZIP_EXE}" "eprosima-fast-rtps.zip" -d "${PROJECT_BINARY_DIR}/doc/"
+#            COMMAND ${CMAKE_COMMAND} -E remove_directory "${PROJECT_BINARY_DIR}/doc/manual"
+#            COMMAND ${CMAKE_COMMAND} -E rename "${PROJECT_BINARY_DIR}/doc/eprosima-fast-rtps-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual"
+#            COMMAND ${CMAKE_COMMAND} -E remove "eprosima-fast-rtps.zip"
```
Comment (MiguelCompany) [inline CMakeLists.txt:275]:
```suggestion
#    if(NOT CHECK_DOCUMENTATION)
#        add_custom_target(readthedocs
#            COMMAND "${WGET_EXE}" "https://media.readthedocs.org/htmlzip/eprosima-fast-cdr/v${PROJECT_VERSION}/eprosima-fast-cdr.zip"
#            COMMAND "${UNZIP_EXE}" "eprosima-fast-cdr.zip" -d "${PROJECT_BINARY_DIR}/doc/"
#            COMMAND ${CMAKE_COMMAND} -E remove_directory "${PROJECT_BINARY_DIR}/doc/manual"
#            COMMAND ${CMAKE_COMMAND} -E rename "${PROJECT_BINARY_DIR}/doc/eprosima-fast-cdr-v${PROJECT_VERSION}" "${PROJECT_BINARY_DIR}/doc/manual"
#            COMMAND ${CMAKE_COMMAND} -E remove "eprosima-fast-cdr.zip"
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]
Avoid unnamed parameters and use `(void) param;` trick instead, so the parameters can be documented.

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/38/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/39/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Linux/39/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastCDR%20Manual%20Windows/40/)

**Comment by MiguelCompany** [comment]
Uncrustify being handled on #79

**Review by MiguelCompany — APPROVED** [review]
LGTM

**Pull request merged**